Assets/MixedRealityToolkit/Providers/BaseSpatialObserver.cs

```diff
@@ -55,6 +55,25 @@ public abstract class BaseSpatialObserver : BaseDataProvider<IMixedRealitySpatia
         /// </summary>
         protected IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem { get; private set; }
 
+        /// <inheritdoc />
+        public override void Enable()
+        {
+            if (!IsRunning && StartupBehavior == AutoStartBehavior.AutoStart)
```

Comment:
The AutoStart behavior is handled in Iniitalize in both of the derived classes:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/7c6b2907cd20d9aa8880bf4e99b457d3478a54e8/Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/SpatialObjectMeshObserver.cs#L111

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/7c6b2907cd20d9aa8880bf4e99b457d3478a54e8/Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs#L420

We should keep the `autoResume`  behavior I commented on above here.
The intent was:

1. Initialize -> autostart based on `StartupBehavior`
2. Disable -> set `autoResume` if currently running
3. Enable -> restart if `autoResume` is set

Reply:
hmmm is it not better to use StartupBehaviour as the autoResume? 

Like when I re-enable, if I am auto-start, then I should automatically start the feature. This way users could also tune observers to switch between auto-start and manual-start after enables/disables.

Whereas with autoResume based on last state in disable, StartupBehaviour is use-less after initialize. The user can't change the functionality of autoResume unless they suspend the observer before every disable(). 

Compared to StartupBehaviour  is our indicator from the user if we should auto-resume.

Thoughts?

Reply:
Although also see #6920 which is somewhat related here

Reply:
It's a bit of a change in behavior, but yeah I think I agree with you that this way is more configurable and better.

Reply:
Please add documentation to the release notes / updating guide as part of this PR. Otherwise, the behavior change may get lost. Thanks!